application/lib/fluent_url.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+require 'addressable/uri'
+
+class FluentUrl
+  def initialize(base)
+    @uri = Addressable::URI.parse(base)
+    @segments = []
+    @params = {}
+  end
+
+  def add_path(part)
+    return self if part.blank?
+
+    @segments << part
+    self
+  end
+
+  def add_param(key, value)
+    @params[key.to_s] = value
```

Comment:
don't you need to escape the values for the query string, so that if I call `add_param(:search, 'sea biology')` it is escaped into `sea+biology` for example?

Reply:
yes, but this is done by the Addressable lib automatically. This is one of the reasons to use it